### PR TITLE
Add transfer progress

### PR DESCRIPTION
**Pull request by JonasVautherin**

### CODE REVIEW

File: protos/mission/mission.proto [hunk 1]
```diff
@@ -221,3 +249,17 @@ message MissionResult {
     Result result = 1; // Result enum value
     string result_str = 2; // Human-readable English string describing the result
 }
+
+// Progress data coming from mission upload, in percent.
+message ProgressData {
+    float progress = 1 [(mavsdk.options.default_value)="NaN"]; // Progress (percentage)
```
Comment (julianoes) [inline protos/mission/mission.proto:255]: Percentage would mean 0..100 but I'd probably just do 0..1.0, right?

File: protos/mission/mission.proto [hunk 2]
```diff
@@ -221,3 +249,17 @@ message MissionResult {
...
+}
+
+// Progress data coming from mission download, in percent, or the mission itself (if the transfer succeeds).
+message ProgressDataOrMission {
+    bool has_progress = 1 [(mavsdk.options.default_value)="false"]; // Whether this ProgressData contains a 'progress' status or not
+    float progress = 2 [(mavsdk.options.default_value)="NaN"]; // Progress (percentage)
+
+    bool has_mission_plan = 3; // Whether this ProgressData contains a 'mission_plan' or not
+    MissionPlan mission_plan = 4; // Mission plan
+}
```
Comment (julianoes) [inline protos/mission/mission.proto:265]: I'm confused, this is not used anywhere...

File: protos/mission/mission.proto (hunk 1 above)
Comment (JonasVautherin) [reply protos/mission/mission.proto:255]: This is the same as the calibration, but I really don't mind. I'll change it to `Progress (0..1.0)` :+1:

**Review by julianoes — DISMISSED** [review]
Yes, now it makes more sense!

**Review by julianoes — APPROVED** [review]

**Pull request merged**